site/src/_pages/spec/1_block-types.mdx

```diff
@@ -72,18 +108,37 @@ Blocks should expect the following additional fields related to linked data to b
 
 - a `linkGroups` field which provides the links attached to the block or entities provided in either `linkedEntities` or `linkedAggregations`, grouped by entity and path
 
-- where available, an `entityTypes` field containing entity type definitions for any entities sent to the blocks, which can be parsed to understand the constraints on user input for each entity (e.g. which fields are editable, what sort of data they accept).
-  Each entry in the array is a JSON schema object with an additional `entityTypeId` field corresponding to the `entityTypeId` of the entities it describes the shape of.
-
 _\* Embedding applications will want to impose limits on the extent to which links between their records are followed for any given query, and not all indirectly linked entities may be provided with the block’s initial data.
 They can be requested using the functions listed below._
 
-Each entity provided under `linkedEntities` or `linkedAggregations` will also have an `entityId` and `entityTypeId` to be used as arguments when updating those entities.
+Each entity provided under `linkedEntities` or `linkedAggregations` will also have [identifying fields](#fields-identifying-blocks) to be used as arguments when updating those entities.
+
 See [linking entities](#linking-entities) for a discussion of how links are managed.
 
+#### Type fields
+
+Where available, blocks SHOULD expect and handle an `entityTypes` field containing entity type definitions for any entities sent to the blocks, which can be parsed to understand the constraints on user input for each entity (e.g. which fields are editable, what sort of data they accept).
+
+Each entry in the `entityTypes` array is a JSON schema object with an additional `entityTypeId` field corresponding to the `entityTypeId` of the entities it describes the shape of (which MAY be different to the URI value for the standard JSON schema `$id` property).
+
 ### Entity functions
 
-Subject to the permissions granted to them by the embedding application, blocks can expect functions with the names and signatures listed below to be made available to them, i.e. to be passed in along with the properties defined in their schema, or to be otherwise made available in their scope depending on their [implementation](#linking-entities):
+Subject to [the permissions granted to them by the embedding application](#block-permissions), blocks can expect functions with the names and signatures listed below to be made available to them, i.e. to be passed in along with the properties defined in their schema, or to be otherwise made available in their scope depending on their [implementation](https://blockprotocol.org/spec/implementation-approaches).
+
+Blocks should have sensible fallbacks for when permissions are denied them, or when functions are absent, for example:
+
+- implementing a readonly / display mode for data which they are unable to edit
```

Comment:
Do we want to say "e.g. when a text block is rendered as part of a static page, as opposed to within a content management system or editing environment"?

Reply:
Have done something like this